data_steward/cdr_cleaner/cleaning_rules/convert_pre_post_coordinated_concepts.py

```diff
@@ -0,0 +1,256 @@
+INSERT INTO `{{project}}.{{dataset}}.observation`
+({{observation_fields}})
+SELECT 
+    ROW_NUMBER() OVER(
+        PARTITION BY o.observation_id
+        ORDER BY
+            m.new_observation_concept_id, 
+            m.new_value_as_cocnept_id
+        ) * 100000000000 + o.observation_id AS observation_id,
+    o.person_id,
+    m.new_observation_concept_id AS observation_concept_id,
+    o.observation_date,
+    o.observation_datetime,
+    o.observation_type_concept_id,
+    o.value_as_number,
+    o.value_as_string,
+    m.new_value_as_cocnept_id AS value_as_concept_id,
+    o.qualifier_concept_id,
+    o.unit_concept_id,
+    o.provider_id,
+    o.visit_occurrence_id,
+    o.visit_detail_id,
+    o.observation_source_value,
+    o.observation_source_concept_id,
+    o.unit_source_value,
+    o.qualifier_source_value,
+    o.value_source_concept_id,
+    o.value_source_value, 
+    o.questionnaire_response_id
+FROM `{{project}}.{{sandbox_dataset}}.{{sandbox_table}}` o
+JOIN `{{project}}.{{dataset}}.concept` c
+ON o.value_source_value = c.concept_code
+JOIN `{{project}}.{{sandbox_dataset}}.{{sandbox_table}}_mapping` m
+ON c.concept_id = m.concept_id
+""")
+
+
+class ConvertPrePostCoordinatedConcepts(BaseCleaningRule):
+
+    def __init__(self,
+                 project_id,
+                 dataset_id,
+                 sandbox_dataset_id,
+                 table_namer=None):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = (
+            'Converts the observation records with pre-coordinated mapping to'
+            'post-coordinated mapping.')
+
+        super().__init__(issue_numbers=JIRA_ISSUE_NUMBERS,
+                         description=desc,
+                         affected_datasets=[cdr_consts.RDR],
+                         affected_tables=[OBSERVATION],
+                         project_id=project_id,
+                         dataset_id=dataset_id,
+                         sandbox_dataset_id=sandbox_dataset_id,
+                         depends_on=[SetConceptIdsForSurveyQuestionsAnswers],
+                         table_namer=table_namer)
+
+    def get_query_specs(self, *args, **keyword_args) -> query_spec_list:
+        """
+        Queries are generated for creating a sandbox table, deleting records from observation,
+        and inserting new records to observation.
+
+        The sandbox query sandboxes rows from observation that meet the following criteria:
+            1. value_source_concept_id is a non-standard concept, and
+            2. value_source_value has a "Maps to value" relationship.
+
+        The delete query deletes rows from observation that are sandboxed.
+
+        The insert query isnerts rows with newly assigned observation_ids and updated
```

Comment:
These are great descriptions! Thanks for making these observations distinguishable from the backfill_skip observations as well as making it all clear in the note. 

There is a tiny typo here.

Reply:
Thanks! I fixed the typo.